app.py

```diff
@@ -400,9 +416,24 @@ def utility():
     STATUS_CREATE = "create"
     STATUS_CHECK = "check"
 
+    if not bronze.authorized:
+        return redirect(url_for('oauth'))
+
+    url = app.config["BRONZE_ADDRESS"] + '/Manage/Kyc'
```

Comment:
move oauth resources (url and userinfo etc) into the global context, something like:

```
@app.before_request
def before_request_func():
    g.bronze_kyc_url = XXXX
    g.bronze_userinfo = XXXX
```

This way we can set it once and it can be used in every template via the parent layout.html template

Reply:
changed this a bit.